instrumentation/opentelemetry-instrumentation-asgi/src/opentelemetry/instrumentation/asgi/__init__.py

```diff
@@ -29,21 +29,31 @@
 from opentelemetry import context, propagators, trace
 from opentelemetry.instrumentation.asgi.version import __version__  # noqa
 from opentelemetry.instrumentation.utils import http_status_to_canonical_code
+from opentelemetry.trace.propagation.textmap import DictGetter
 from opentelemetry.trace.status import Status, StatusCanonicalCode
 
 
-def get_header_from_scope(scope: dict, header_name: str) -> typing.List[str]:
-    """Retrieve a HTTP header value from the ASGI scope.
+class CarrierGetter(DictGetter):
```

Comment:
Shouldn't these be extending from `Getter` instead of `DictGetter`?

Reply:
@lzchen Extending DictGetter because of the following reasons:

- carrier is of type dict
- there is no keys implementation required here so we can use the default keys implementation in DictGetter
- if we have to extend the Getter Implementation we have to override the keys() method as well

Reply:
Yea, this getter is more of a specialized implementation of a dict getter so LGTM.